src/dxvk/dxvk_cs.cpp

```diff
@@ -202,11 +202,20 @@ namespace dxvk {
       while (!m_stopped.load()) {
         { std::unique_lock<dxvk::mutex> lock(m_mutex);
 
-          m_condOnAdd.wait(lock, [this] {
-            return (!m_queueOrdered.queue.empty())
-                || (!m_queueHighPrio.queue.empty())
-                || (m_stopped.load());
-          });
+          if (unlikely(m_queueOrdered.queue.empty()
+            && m_queueHighPrio.queue.empty()
+            && (!m_stopped.load()))) {
```

Comment:
could make this a bit more elegant and do something like
```c++
auto pred = [this] { return !queue.empty() && ... };

if (!pred()) {
  ...
  m_condOnAdd.wait(lock, pred)
}
```

since the condition isn't trivial her